engines/ags/lib/alfont/alfont.cpp

```diff
@@ -60,7 +60,11 @@ size_t alfont_text_height(ALFONT_FONT *font) {
 
 void alfont_textout(BITMAP *bmp, ALFONT_FONT *font, const char *text, int x, int y, uint32 color) {
 	Graphics::ManagedSurface &surf = **bmp;
-	font->getFont()->drawString(&surf, text, x, y, bmp->w - x, color);
+	Graphics::Font *fnt = font->getFont();
+
+	if (surf.hasTransparentColor())
+		fnt->setTransparentColor(surf.getTransparentColor());
+	fnt->drawString(&surf, text, x, y, bmp->w - x, color);
```

Comment:
That's actually an excellent idea. It would be cleaner than having an explicit setTransparentColor call. Graphics::Font can have a version that simply passes on the internal surface to the existing version, and the TTFFont can overload it to get the transparent color. I'll rework the PR tonight.